nservicebus/msmq/scalability-and-ha/sender-side-distribution.md

```diff
@@ -0,0 +1,120 @@
+
+NOTE: Sender-side distribution is available in NServiceBus Versions 6 and above.
+
+When using sender-side distribution:
+
+* Multiple endpoint instances (deployed to different servers) are capable of processing a message that requires scaled-out processing.
+* A client sending a message is aware of all the endpoint instances that can process the message.
+* The client sends the message to a worker endpoint instance based on round-robin distribution, or a custom distribution strategy.
+
+Using sender-side distribution requires two parts. The first part maps message types to logical endpoints, and occurs in code. The second part maps logical endpoints to physical endpoint instances running on a specific machine.
+
+
+## Mapping logical endpoints
+
+To map message types to logical endpoints, use the following config:
+
+snippet:Routing-FileBased-Config
+
+This creates mappings specifying that the `AcceptOrder` command is handled by the **Sales** endpoint, while the `SendOrder` command is handled by the **Shipping** endpoint.
+
+Meanwhile, the logical-to-physical mappings will be configured in the Routes.xml file, as this information is an operational concern that must be changed for deployment to multiple machines.
+
+WARNING: If a message is mapped in an App.config file via the `UnicastBusConfig/MessageEndpointMappings` configuration section, then that message cannot participate in sender-side distribution. The endpoint address specified by a message endpoint mapping is a physical address (`QueueName@MachineName`, where machine name is assumed to be `localhost` if omitted) which combines the message-to-owner-endpoint and endpoint-to-physical-address concerns in a way that can't be separated.
+
+
+## Mapping physical endpoint instances
+
+The routing configuration file specifies how logical endpoint names are mapped to physical queues on specific machines:
+
+snippet:Routing-FileBased-MSMQ
+
+By default, a round-robin distribution strategy is used to distribute messages between the available endpoint instances.
+
+
+## Mapping file
+
+The routing file is a simple XML file that has to be located either on a local hard drive or a network drive. When using MSMQ as the transport, NServiceBus will automatically look for an `instance-mapping.xml` file in `AppDomain.BaseDirectoy`.
+
+The mapping file is processed before the endpoint starts up and then re-processed at regular intervals so the changes in the document are reflected in the behavior of NServiceBus automatically. If the document is not present in its configured location when endpoint starts up, NServiceBus will refuse to run. If it is deleted when the endpoint is already running, it will continue to run normally (with the last successfully read routes).
```

Comment:
I've checked in code and this has been changed in https://github.com/Particular/NServiceBus/pull/3900 so that if the instance mapping file is not present, it will be ignored and no routing configuration will be applied. What's more, the location will not be monitored so when a file is created when the endpoint is running, it won't be picked up.